src/ASM/DualField.C

```diff
@@ -64,6 +108,17 @@ bool DualRealFunc::inDomain (const Vec3& X) const
     return y <= 0.5*width && y >= -0.5*width;
   }
 
+  if (patch && !Delem.empty())
+  {
+    const Vec4* Xp = dynamic_cast<const Vec4*>(&X);
```

Comment:
should cast be checked?

Reply:
Ideally yes, but it would be a logic error if a non-Vec4 object is provided here (or the u-member is null), and I just wanted it to seg-fault in those cases to easily catch them. But maybe in debug mode.